pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -341,6 +344,11 @@ func translateRoute(in *networking.HTTPRoute,
 			}
 		}
 	}
+
+	if fault := in.Fault; fault != nil {
+		out.PerFilterConfig[xdsutil.Fault] = util.BuildProtoStruct(translateFault(in.Fault))
```

Comment:
`util.BuildProtoStruct` creates the struct inside "value" field, which produces the following configuration:
```
 "per_filter_config": {
          "envoy.fault": {
                   "value": {
                            "delay": {
                                "FaultDelaySecifier": {
                                       "FixedDelay": 7000000000
                                 },
                                 "percent": 100
                            }
                    }
        }
},
```

The "value" struct is redundant, its contents should appear immediately in the "envoy.fault" struct.

This causes the following error in Envoy:
`Error adding/updating listener http_0.0.0.0_9080: Unable to parse JSON as proto (INVALID_ARGUMENT:: Cannot find field.): {\"value\":{\"delay\":{\"FaultDelaySecifier\":{\"FixedDelay\":7000000000},\"percent\":100}}}"`

Reply:
Excellent find!!

Reply:
Found the problem, that line should be:
```
out.PerFilterConfig[xdsutil.Fault] = util.MessageToStruct(translateFault(in.Fault))
```